hazelcast-client/src/main/java/com/hazelcast/client/proxy/ClientMapProxy.java

```diff
@@ -1361,21 +1362,21 @@ public void putAll(Map<? extends K, ? extends V> m) {
             Data keyData = toData(entry.getKey());
 
             int partitionId = partitionService.getPartitionId(keyData);
-            Map<Data, Data> partition = entryMap.get(partitionId);
+            List<Map.Entry<Data, Data>> partition = entryMap.get(partitionId);
             if (partition == null) {
-                partition = new HashMap<Data, Data>();
+                partition = new LinkedList<Entry<Data, Data>>();
```

Comment:
Given http://stackoverflow.com/questions/322715/when-to-use-linkedlist-over-arraylist , would it make better sense to use ArrayList here? I get that if the backing array has to be resized we incur a time penalty, but that may be mitigated by estimating the size needed on construction.